contrib/opencensus-ext-requests/tests/test_requests_trace.py

```diff
@@ -204,6 +217,144 @@ def test_wrap_requests_exporter_thread(self):
+        }
+        expected_name = '/test'
+        expected_status = status_module.Status(4, 'request timed out')
+
+        self.assertEqual(span_module.SpanKind.CLIENT,
+                         mock_tracer.current_span.span_kind)
+        self.assertEqual(expected_attributes,
+                         mock_tracer.current_span.attributes)
+        self.assertEqual(expected_name, mock_tracer.current_span.name)
+        self.assertEqual(
+            expected_status.__dict__,
+            mock_tracer.current_span.status.__dict__
+        )
+
+    def test_wrap_requests_invalid_url(self):
+        mock_return = mock.Mock()
+        mock_return.status_code = 200
+        return_value = mock_return
+        mock_func = mock.Mock()
+        mock_func.__name__ = 'get'
+        mock_func.return_value = return_value
+        mock_func.side_effect = requests.URLRequired
+        mock_tracer = MockTracer()
+
+        patch = mock.patch(
+            'opencensus.ext.requests.trace.execution_context.'
+            'get_opencensus_tracer',
+            return_value=mock_tracer)
+
+        patch_thread = mock.patch(
+            'opencensus.ext.requests.trace.execution_context.'
+            'is_exporter',
+            return_value=False)
+
+        wrapped = trace.wrap_requests(mock_func)
+
+        url = 'http://localhost:8080/test'
+
+        with patch, patch_thread:
+            wrapped(url)
+
+        expected_attributes = {
+            'http.host': 'localhost:8080',
+            'http.method': 'GET',
+            'http.path': '/test',
+            'http.url': url,
+        }
+        expected_name = '/test'
+        expected_status = status_module.Status(3, 'invalid URL')
+
+        self.assertEqual(span_module.SpanKind.CLIENT,
+                         mock_tracer.current_span.span_kind)
+        self.assertEqual(expected_attributes,
+                         mock_tracer.current_span.attributes)
+        self.assertEqual(expected_name, mock_tracer.current_span.name)
+        self.assertEqual(
+            expected_status.__dict__,
+            mock_tracer.current_span.status.__dict__
+        )
+
+    def test_wrap_requests_exception(self):
+        mock_return = mock.Mock()
+        mock_return.status_code = 200
+        return_value = mock_return
+        mock_func = mock.Mock()
+        mock_func.__name__ = 'get'
+        mock_func.return_value = return_value
+        mock_func.side_effect = requests.TooManyRedirects
+        mock_tracer = MockTracer()
+
+        patch = mock.patch(
+            'opencensus.ext.requests.trace.execution_context.'
+            'get_opencensus_tracer',
+            return_value=mock_tracer)
+
+        patch_thread = mock.patch(
+            'opencensus.ext.requests.trace.execution_context.'
+            'is_exporter',
+            return_value=False)
+
+        wrapped = trace.wrap_requests(mock_func)
+
+        url = 'http://localhost:8080/test'
+
+        with patch, patch_thread:
+            wrapped(url)
```

Comment:
To be clearer, given `mock_func` raise an exception, this test case should fail gith here and never go further. The test should actually assert the `requests.TooManyRedirects` exception is raised from `wrapped(url)`.